### PR TITLE
Implement SPV_INTEL_fpga_cluster_attributes extension

### PR DESCRIPTION
**Pull request by MrSidims**
This extension adds a decoration, applicable on functions, that
statically-scheduled clusters should handle stalls using a
stall-enable signal to freeze computation within the cluster.

Spec: https://github.com/KhronosGroup/SPIRV-Registry/pull/85/

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
- [ ] Propagate to SPIR-V Headers

**Comment by MrSidims** [comment]
@AlexeySotkin @AlexeySachkov  please take a look

File: lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -4169,6 +4170,16 @@ bool SPIRVToLLVM::transVectorComputeMetadata(SPIRVFunction *BF) {
   return true;
 }
 
+bool SPIRVToLLVM::transFPGAFunctionMetadata(SPIRVFunction *BF, Function *F) {
+  if (BF->hasDecorate(DecorationStallEnableINTEL)) {
+    std::vector<Metadata *> MetadataVec;
+    MetadataVec.push_back(ConstantAsMetadata::get(
+        ConstantInt::get(Type::getInt32Ty(*Context), 1)));
```
Comment (AlexeySotkin) [inline lib/SPIRV/SPIRVReader.cpp:4177]:
```suggestion
        getInt32Ty(M, 1));
```

**Review by AlexeySotkin — APPROVED** [review]
Just one minor comment. Overall LGTM. Thanks!

File: lib/SPIRV/SPIRVReader.cpp (hunk 1 above)
Comment (MrSidims) [reply lib/SPIRV/SPIRVReader.cpp]: Done, thanks!

**Comment by AlexeySotkin** [comment]
@mlychkov please take a look.

**Review by mlychkov — APPROVED** [review]

**Review by AlexeySotkin — APPROVED** [review]

**Pull request merged**